### PR TITLE
Feat/4732  improve link management

### PR DESCRIPTION
**Pull request by charline-rageade**
#4732 

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [ ] Manual, quality tests have been done

### CODE REVIEW

File: frontend_lib/src/helper.js [hunk 1]
```diff
@@ -799,3 +799,27 @@ export const getCoverBaseUrl = (apiUrl, userId) => `${apiUrl}/users/${userId}/co
...
+
+export const addExternalLinksIcons = (htmlString) => {
+  const doc = getDocumentFromHTMLString(htmlString)
+  const locationUrl = new URL(location.toString())
+  for (const link of doc.getElementsByTagName('a')) {
+    const url = new URL(link.href)
+    if (url.origin === locationUrl.origin) continue
+    const icon = doc.createElement('i')
+    icon.className = 'fas fa-external-link-alt'
+    icon.style = 'margin-inline-start :5px;'
```
Comment (gbonaspetti) [inline frontend_lib/src/helper.js:811]:
```suggestion
    icon.style = 'margin-inline-start: 5px;'
```
no?

File: frontend_lib/src/helper.js [hunk 2]
```diff
@@ -799,3 +799,27 @@ export const getCoverBaseUrl = (apiUrl, userId) => `${apiUrl}/users/${userId}/co
 export const getFileDownloadUrl = (apiUrl, workspaceId, contentId, filename) => `${apiUrl}/workspaces/${workspaceId}/files/${contentId}/raw/${filename}?force_download=1`
 
 export const htmlToText = (domParser, htmlString) => domParser.parseFromString(htmlString, 'text/html').documentElement.textContent
+
+export const addExternalLinksIcons = (htmlString) => {
+  const doc = getDocumentFromHTMLString(htmlString)
+  const locationUrl = new URL(location.toString())
```
Comment (raphj) [inline frontend_lib/src/helper.js:805]: If you get a linting error here, you may be able to fix it like this
```suggestion
  const locationUrl = new URL(window.location.toString())
```

File: frontend_lib/src/helper.js [hunk 3]
```diff
@@ -799,3 +799,27 @@ export const getCoverBaseUrl = (apiUrl, userId) => `${apiUrl}/users/${userId}/co
...
+    link.appendChild(icon)
+  }
+  return doc.body.innerHTML
+}
+
+export const getDocumentFromHTMLString = (htmlString) => {
+  const doc = new DOMParser().parseFromString(htmlString, 'text/html')
+
+  if (doc.documentElement.tagName === 'parsererror') {
+    throw new Error('Cannot parse string: ' + doc.documentElement.textContent)
```
Comment (raphj) [inline frontend_lib/src/helper.js:821]: Here, you may prefer to use console.error and return an empty string. Exceptions can lead to a blank page on Tracim. Maybe see this with @Skylsmoi

**Review by raphj — APPROVED** [review]

File: frontend_lib/src/helper.js (hunk 3 above)
Comment (charline-rageade) [reply frontend_lib/src/helper.js:821]: This code is older and is not part of the modification of this feature !

File: frontend_lib/src/helper.js (hunk 1 above)
Comment (charline-rageade) [reply frontend_lib/src/helper.js:811]: fixed

File: frontend_lib/src/helper.js (hunk 2 above)
Comment (charline-rageade) [reply frontend_lib/src/helper.js:805]: fixed

File: frontend_lib/test/apiMock.js [hunk 4]
```diff
@@ -3,6 +3,13 @@ import { author } from './fixture/author.js'
 
 const nock = require('nock')
 
+const mockGetContentWithoutWorkspaceId200 = (apiUrl, contentId) => {
+  console.log('mock called with', contentId)
```
Comment (gbonaspetti) [inline frontend_lib/test/apiMock.js:7]: forgotten console log?

**Review by gbonaspetti — CHANGES_REQUESTED** [review]

File: frontend_lib/test/apiMock.js (hunk 4 above)
Comment (charline-rageade) [reply frontend_lib/test/apiMock.js]: fixed

**Review by gbonaspetti — APPROVED** [review]

**Pull request merged**